### ISSUE TITLE
Constructor.getAnnotatedParameterTypes throws exception when trying to get annotated parameter type in an enum

### ISSUE THREAD

**Issue by felipepontes-research** [eclipse/openj9#1851]
An exception is thrown when executing the following program:

```
public enum A {}

import java.lang.reflect.Constructor;
import java.lang.reflect.AnnotatedType;
public class B {
   public static void main(String[] args) {
      for (Constructor c : A.class.getDeclaredConstructors()) {
         for (AnnotatedType at : c.getAnnotatedParameterTypes()) {
            System.out.println(at.getType());
         }
      }
   }
}
```

**affected versions:**

openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-b12)
Eclipse OpenJ9 VM (build openj9-0.8.0, JRE 1.8.0 Linux amd64-64 Compressed References 20180315_120 (JIT enabled, AOT enabled)
OpenJ9 - e24e8aa9
OMR - 3e8296b4
JCL - ee1e77df1d based on jdk8u162-b12)

**steps to reproduce:**

	1. compile above program
	2. run it using openj9 java

**current result:**

Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Array index out of range: 0
	at java.lang.reflect.Executable.getAllGenericParameterTypes(Executable.java:318)
	at java.lang.reflect.Executable.getAnnotatedParameterTypes(Executable.java:693)
	at B.main(B.java:4)

**expected result:**

class java.lang.String
int

**Comment by JasonFengJ9** [comment]
Not able to reproduce the exception reported.

@felipepontes-research could you provide the actual `java` file and command line options? thanks.

**Comment by felipepontes-research** [comment]
Please, use the following files: [sample.zip](https://github.com/eclipse/openj9/files/1981145/sample.zip)
I'm not using any command line options (i.e. ```jdk8u162-b12_openj9-0.8.0/bin/java B```).

**Comment by JasonFengJ9** [comment]
Thanks, the exception reproduced.

**Comment by JasonFengJ9** [comment]
This should be closed by #2053 (the clause closing multiple issues wasn't in proper format).
